subprojects/core/src/main/java/org/gradle/cache/internal/DefaultCacheScopeMapping.java

```diff
@@ -42,7 +45,7 @@ public DefaultCacheScopeMapping(File userHomeDir, @Nullable File projectCacheDir
 
     @Override
     public File getBaseDirectory(@Nullable Object scope, String key, VersionStrategy versionStrategy) {
-        if (key.equalsIgnoreCase("projects") || key.equalsIgnoreCase("tasks") || !key.matches("\\p{Alpha}+[-//.\\w]*")) {
+        if (key.equalsIgnoreCase("projects") || key.equalsIgnoreCase("tasks") || !key.matches("\\p{Alpha}+[-/.\\w]*")) {
```

Comment:
Would make sense to extract that pattern into a constant.

Reply:
Extracted.